internal/envoy/v3/stats.go

```diff
@@ -41,41 +42,46 @@ func StatsListeners(metrics contour_api_v1alpha1.MetricsConfig, health contour_a
 	// Create HTTPS listener for metrics and HTTP listener for health.
 	case metrics.TLS != nil:
 		listeners = []*envoy_listener_v3.Listener{{
-			Name:          "stats",
-			Address:       SocketAddress(metrics.Address, metrics.Port),
-			SocketOptions: TCPKeepaliveSocketOptions(),
+			Name:                          "stats",
+			Address:                       SocketAddress(metrics.Address, metrics.Port),
+			PerConnectionBufferLimitBytes: &wrappers.UInt32Value{Value: 32768},
```

Comment:
should be able to do this instead with the package helper function(s) (and for other places in this PR)
```suggestion
			PerConnectionBufferLimitBytes: wrapperspb.UInt32(32768),
```